endpoints/openrtb2/auction.go

```diff
@@ -301,6 +302,17 @@ func (deps *endpointDeps) validateRequest(req *openrtb.BidRequest) []error {
 		return errL
 	}
 
+	ccpaPolicy, ccpaPolicyErr := ccpa.ReadPolicy(req)
+	if ccpaPolicyErr != nil {
+		errL = append(errL, ccpaPolicyErr)
+		return errL
+	}
+
+	if err := ccpaPolicy.Validate(); err != nil {
+		errL = append(errL, err)
+		return errL
+	}
+
```

Comment:
You're right. I consider this part of 'transition pain'. I'm trying to keep all privacy logic in each respective privacy policy and will move the GDPR and COPPA logic there too, but that's a bit too much for this PR. Do you think this will cause a major performance issue? I could restructure a few things for now to avoid the extra unmarshal if you feel it's better.

Reply:
No, I'll think it'd be perfectly fine to refactor in another PR latter

Reply:
![image](https://user-images.githubusercontent.com/15863840/70823632-5a9d6180-1dae-11ea-9649-2ee1a7c975be.png)
Can we write a test case to cover `ReadPolicy(req)` error scenario?

Reply:
Nope. That code isn't reachable. It will only get triggered if the `regs.ext` is malformed and there' s already a check for that earlier in the func. File this under pain of a partial refactor of privacy things. More to come soon.